src/cli/cli.cpp

```diff
@@ -4850,7 +4847,8 @@ otError Interpreter::ProcessNetworkDiagnostic(uint8_t aArgsLength, char *aArgs[]
 
     if (strcmp(aArgs[0], "get") == 0)
     {
-        IgnoreError(otThreadSendDiagnosticGet(mInstance, &address, tlvTypes, count));
+        SuccessOrExit(
+            error = otThreadSendDiagnosticGet(mInstance, &address, tlvTypes, count, HandleDiagnosticGetResponse, this));
```

Comment:
In most places, the style we use `&Interpreter::HandleDiagnosticGetResponse` rather than just `HandleDiagnosticGetResponse`. Thoughts?